buffer/src/main/java/io/netty/buffer/PooledByteBufAllocator.java

```diff
@@ -343,36 +342,36 @@ public void freeThreadLocalCache() {
     }
 
     final class PoolThreadLocalCache extends FastThreadLocal<PoolThreadCache> {
-        private final AtomicInteger index = new AtomicInteger();
-        final AtomicInteger caches = new AtomicInteger();
 
         @Override
-        protected PoolThreadCache initialValue() {
-            caches.incrementAndGet();
-            final int idx = index.getAndIncrement();
-            final PoolArena<byte[]> heapArena;
-            final PoolArena<ByteBuffer> directArena;
-
-            if (heapArenas != null) {
-                heapArena = heapArenas[Math.abs(idx % heapArenas.length)];
-            } else {
-                heapArena = null;
-            }
+        protected synchronized PoolThreadCache initialValue() {
```

Comment:
@Scottmitch not sure :-). let me think about it. The `PoolArena.numThreadCache` variables are incremented in the `PoolThreadCache` constructor. Have to think about if the algorithm may break, if this is not done as a unit.
@normanmaurer hmm. this code only runs once per thread on the first allocation. I am not sure why these cases are relevant, or would help on code that's executed so infrequently? I'd prefer to have clean & simple code to be honest.